### PR TITLE
refactor: Remove deprecated endpoints, methods and properties.

### PR DESCRIPTION
**Pull request by ojongerius**
Removes bits that were marked as deprecated a while ago, as far as I can tell they were no longer in use. Tests still pass.

```
  total:     1343
  passing:   1343
  duration:  12.1s
```

/cc @BerkeleyTrue

### CODE REVIEW

**Comment by BerkeleyTrue** [comment]
Are these in production?

**Comment by ojongerius** [comment]
`/modern-challenge-completed` definitely was. I was hoping you would have more confidence in what is out there at the moment -I have no insight in how you deploy FCC.

File: common/models/user.json [hunk 1]
```diff
@@ -148,10 +148,6 @@
       "description": "The challenge last visited by the user",
       "default": ""
     },
-    "currentChallenge": {
```
Comment (raisedadead) [inline common/models/user.json]: We still need this, because of our legacy database. This will mutate the User Object (old) in a way that we do not want.

**Review by raisedadead — CHANGES_REQUESTED** [review]

**Comment by raisedadead** [comment]
Hi @ojongerius I am closing this for now, and lets revisit the cleanup once again.

**Pull request closed without merging**